XUtils.h

```diff
@@ -41,26 +41,26 @@ static inline bool String_eq(const char* s1, const char* s2) {
    return strcmp(s1, s2) == 0;
 }
 
-char* String_cat(const char* s1, const char* s2);
+char* String_cat(const char* s1, const char* s2) ATTR_MALLOC;
 
-char* String_trim(const char* in);
+char* String_trim(const char* in) ATTR_MALLOC;
 
-char** String_split(const char* s, char sep, size_t* n);
+char** String_split(const char* s, char sep, size_t* n) ATTR_MALLOC;
 
 void String_freeArray(char** s);
 
-char* String_getToken(const char* line, unsigned short int numMatch);
+char* String_getToken(const char* line, unsigned short int numMatch) ATTR_MALLOC;
```

Comment:
Violates no-aliasing requirement.

Reply:
It does return a new copy
https://github.com/htop-dev/htop/blob/be568b1153206a067fd09c0f7d2dcea3d7dbaa68/XUtils.c#L143